server/plugin/api.go

```diff
@@ -1258,23 +1471,295 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+			p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Issues are disabled on this repository.", StatusCode: http.StatusMethodNotAllowed})
+			return
+		}
+
+		c.Log.WithError(err).Warnf("Failed to update the issue")
+		p.writeAPIError(w, &APIErrorResponse{
+			ID: "",
+			Message: "failed to update issue: " + getFailReason(resp.StatusCode,
+				issue.Repo,
+				currentUser.Username,
+			),
+			StatusCode: resp.StatusCode,
+		})
+		return
+	}
+
+	rootID := issue.PostID
+	channelID := issue.ChannelID
+	message := fmt.Sprintf("Updated GitHub issue [#%v](%v)", result.GetNumber(), result.GetHTMLURL())
+	if post != nil {
+		if post.RootId != "" {
+			rootID = post.RootId
+		}
+		channelID = post.ChannelId
+		message += fmt.Sprintf(" from a [message](%s)", permalink)
+	}
+
+	reply := &model.Post{
+		Message:   message,
+		ChannelId: channelID,
+		RootId:    rootID,
+		UserId:    c.UserID,
+	}
+
+	if post != nil {
+		_, appErr = p.API.CreatePost(reply)
+	} else {
+		p.API.SendEphemeralPost(c.UserID, reply)
+	}
+	if appErr != nil {
+		c.Log.WithError(appErr).Warnf("failed to create notification post")
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to create notification post, postID: " + issue.PostID + ", channelID: " + channelID, StatusCode: http.StatusInternalServerError})
+		return
+	}
+
+	p.updatePost(post, issue, w)
+	p.writeJSON(w, result)
+}
+
+func (p *Plugin) CreateCommentToIssue(c *UserContext, w http.ResponseWriter, comment, owner, repo string, post *model.Post, numberInt int) {
+	currentUsername := c.GHInfo.GitHubUsername
+	permalink := p.getPermaLink(post.Id)
+	issueComment := &github.IssueComment{
+		Body: &comment,
+	}
+	githubClient := p.githubConnectUser(c.Context.Ctx, c.GHInfo)
+
+	result, rawResponse, err := githubClient.Issues.CreateComment(c.Ctx, owner, repo, numberInt, issueComment)
+	if err != nil {
+		statusCode := 500
+		if rawResponse != nil {
+			statusCode = rawResponse.StatusCode
+		}
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to create an issue comment: " + getFailReason(statusCode, repo, currentUsername), StatusCode: statusCode})
+		return
+	}
+	rootID := post.Id
+	if post.RootId != "" {
+		// the original post was a reply
+		rootID = post.RootId
+	}
+	permalinkReplyMessage := fmt.Sprintf("[Comment](%v) attached to GitHub issue [#%v](%v)", permalink, numberInt, result.GetHTMLURL())
+	reply := &model.Post{
+		Message:   permalinkReplyMessage,
+		ChannelId: post.ChannelId,
+		RootId:    rootID,
+		UserId:    c.UserID,
+	}
+
+	_, appErr := p.API.CreatePost(reply)
+	if appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to create notification post " + post.Id, StatusCode: http.StatusInternalServerError})
+		return
+	}
+}
+
+func (p *Plugin) CloseOrReopenIssue(c *UserContext, w http.ResponseWriter, status string, statusReason string, owner string, repo string, post *model.Post, numberInt int) {
```

Comment:
if i have given a comment check that for entire code I will not write same same everywhere